### PR TITLE
[v6r14] Added soft implementation of a Direct Acyclic Graph (DAG)

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

File: Core/Utilities/DAG.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    """ add a node to graph
+    """
+    if nodeName not in self.graph:
+      self.graph[nodeName] = set()
+
+  def addEdge( self, fromNode, toNode ):
+    """ add an edge (checks if both nodes exist)
+    """
+    if fromNode not in self.graph or toNode not in self.graph:
+      if fromNode not in self.graph:
```
Comment (chaen) [inline Core/Utilities/DAG.py]: Aren't you doing twice the exact same tests here ?

File: Core/Utilities/DAG.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        gLogger.error( "Missing node from where the edge start" )
+      if fromNode not in self.graph:
+        gLogger.error( "Missing node to where the edge lands" )
+      return
+
+    for node, toNodes in self.graph.iteritems():
+      #This is clearly not enough to assure that it's really acyclic...
```
Comment (chaen) [inline Core/Utilities/DAG.py]: You might want to look at Tarjan's algorithm ?

**Comment by chaen** [comment]
Out of curiosity, why do you need that for ? State transition ? 
Also, there is no standard library in python that can do that for us ?

File: Core/Utilities/DAG.py (hunk 2 above)
Comment (fstagni) [inline Core/Utilities/DAG.py]: I can make a full one, no problem. Just, I don't think there's the need.

**Comment by fstagni** [comment]
If I look here on GitHub I find some implementation of DAGs. I decided to write this simple one because it's simple enough to write, and for our use case we don't need for the moment anything else.

Regarding use case: there's a requirement in LHCbDIRAC to have workflows non strictly sequential (for production jobs), so I think this may help.

**Comment by chaen** [comment]
The tests are not passing anymore

**Comment by chaen** [comment]
Review OK

**Pull request merged**